powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/replay/DefaultReplayVerificationService.java

```diff
@@ -52,9 +52,15 @@ class DefaultReplayVerificationService implements ReplayVerificationService {
     private final PowerAuthServiceConfiguration config;
 
     @Override
-    public void checkAndPersistUniqueValue(UniqueValueType type, Date requestTimestamp, String ephemeralPublicKey, String nonce, String identifier) throws GenericServiceException {
+    public void checkAndPersistUniqueValue(UniqueValueType type, Date requestTimestamp, String ephemeralPublicKey, String nonce, String identifier, String version) throws GenericServiceException {
         logger.debug("Checking and persisting unique value, request type: {}, identifier: {}", type, identifier);
-        final Date expiration = Date.from(Instant.now().plus(config.getRequestExpirationInMilliseconds(), ChronoUnit.MILLIS));
+        final int requestExpiration;
+        if ("3.2".equals(version)) {
```

Comment:
I would prefer to revert the logic to use the new behavior for all new versions (we can easily list the prior versions). This is something we will for sure forget to edit, which will accidentally extend the validity for 3.2+ releases, and we won't even notice.

Reply:
Ok, I can revert the logic.